jina/helper.py

```diff
@@ -647,6 +647,6 @@ def __init__(self, func):
     def __get__(self, obj, cls):
         if obj is None:
             return self
-
-        value = obj.__dict__[f'CACHED_{self.func.__name__}'] = self.func(obj)
```

Comment:
In `jina/executors/__init__.py` there is some logic related to this that needs to be reviewed maybe?

```python
    def __getstate__(self):
        d = dict(self.__dict__)
        del d['logger']
        for k in self._post_init_vars:
            del d[k]
        cached = [k for k in d.keys() if k.startswith('CACHED_')]
        for k in cached:
            del d[k]
        return d
```

Reply:
or was it the source of the problem? that we should say `if not  k.startswith('CACHED_')`?

Reply:
Good catch! This part of the codes is for dumping `executors` into pickles. We don't want to save the cached variables, e.g. the models. However, using the `CACHED_` prefix leads to the failure of retrieving the variable from the `self.__dict__`.

Reply:
your changes will fail. because you save cached which are not pickable. please revert it back

Reply:
This has been handled by another PR